_posts/2020-10-04-newpipe-0.20.0.md

```diff
@@ -1,154 +1,176 @@
 
-Naturally, since this was such a massive change, several bugs also slipped through. But [@avently](https://github.com/avently), with the help of [@blackbox87](https://github.com/blackbox87), [@Stypox](https://github.com/Stypox) and [@TobiGr](https://github.com/TobiGr) fixed them rapidly as they were reported. 
+This evolution saw evolutionary bugs in its wake. [@avently](https://github.com/avently), with the help of [@blackbox87](https://github.com/blackbox87), [@Stypox](https://github.com/Stypox) and [@TobiGr](https://github.com/TobiGr) swatted as the reports came in. 
 
 ## The Notification is dead, long live the Notification!
 
-If you care about such things, you have likely noticed that NewPipe's notification looks quite different from most media player apps these days. It has a progress bar, buttons in odd places, and a permanent grey colour. This is because NewPipe uses an old API for notifications called RemoteView. This allowed for a lot of customisation, and that was good at the time it was in vogue. However, it has been deprecated and superseded by a more modern API.
+As a pedant, you have noticed NewPipe's notification is different from most other media apps. It has the bar of progress, buttons in odd places, and a permanent, though tepid grey colour. Much of the difference is down to using an old API for notifications called RemoteView, _in the past_!
 
 Enter the **Age of MediaStyle notification**:
 
-This is a special notification type which has been available for media apps to use since Android Lollipop. In Oreo, it gained the ability to be coloured to match the colour theme of the cover art of the music playing. And since Android 10, it has a seekbar built right into it! It is a powerful API that removes the need to use a whole bunch of custom code to achieve what we had so far. This will make it far easier to maintain and improve NewPipe's notification. Win for development! So let's get on with what is new and what has changed.
+This is a special notification type available for media apps since Android (5.0) Lollipop. In Oreo (8.0), it could colour-match its theme to the cover art of the music playing. Since Android 10, it has a SeekBar built in. It is a powerful API removing the need to use a whole bunch of custom code to achieve what was. Easier to maintain and improve. Win!
 
 ### Android 9
 
-First of all, we say "sorry" to our users on Android 9 and below. The progress bar is gone. As we've said above, it will only be shown on Android 10 and above. We're a small team and can't maintain code for the old and new types of notifications simultaneously, so the old one has to go. And not just that; the old notification isn't even supported Android 11 onwards. Bye-bye RemoteView notifications. We shall miss you! :'(
-
-But you get a lot in exchange for that sacrifice!
```

Comment:
Again with the sacrifice. Wasn't good the first time. This is still an apologetic tirade of a blogpost. The reader is meant to feel ambivalent about choices? Nobody is invested in an old way to display notifications. It is a detail.

Reply:
> Again with the sacrifice. Wasn't good the first time. This is still an apologetic tirade of a blogpost.

The first time it was a joke. (Refer to explanation above.)

This time, it is an actual trade-off. This is the first time we apologise, so I don't see how one apology can make it "an apologetic tirade of a blogpost".

> The reader is meant to feel ambivalent about choices?

The reader is given a reason why the progress bar they liked is now gone.

> Nobody is invested in an old way to display notifications.

Reply:
There is some reason for that, the blogpost doesn't help in the matter.